### PR TITLE
Be more explicit about lack of inheritance on `thread_mattr_accessor`

### PR DESCRIPTION
**Pull request by ghiculescu**
Just trying to make the docs as clear as possible to mitigate confusion like https://github.com/rails/rails/issues/43228

### CODE REVIEW

File: activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb [hunk 1]
```diff
@@ -111,13 +111,15 @@ def #{sym}=(obj)
...
   # If a subclass changes the value, the parent class' value is not changed.
-  # Similarly, if the parent class changes the value, the value of subclasses
-  # is not changed.
+  # If the parent class changes the value, the value of subclasses is not changed.
   #
   #   class Customer < Account
   #   end
   #
+  #   Account.user  # => "DHH"
+  #   Current.user  # => nil
```
Comment (p8) [inline activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb:122]: Should this be Customer?
```suggestion
  #   Customer.user  # => nil
```

**Comment by p8** [comment]
Hey @ghiculescu :), do you think we should better explain what sets them apart from `mattr_accessor`?
Maybe mention they are thread local? Or that they use Thread.current?

**Comment by ghiculescu** [comment]
It's mentioned at the top of the file:

image

To me, that doesn't really explain why subclassing doesn't work. So I figure if anyone really understands the internals that should do it. I'm not sure repeating it in more places will change much 🤷

**Comment by p8** [comment]
Ah, I missed that. 👍

**Comment by p8** [comment]
Hmm, because these methods are defined directly on `Module` they get merged with all the other extensions on `Module`:
image
https://edgeapi.rubyonrails.org/classes/Module.html

Maybe we can add a headings above? `== Attribute Accessors`
https://github.com/rails/rails/blob/03bb5c1dc24dc635ab16f8f4a89aef3ff8d2af00/activesupport/lib/active_support/core_ext/module/attribute_accessors.rb#L3
And  `== Attribute Accessors per Thread`
https://github.com/rails/rails/blob/03bb5c1dc24dc635ab16f8f4a89aef3ff8d2af00/activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb#L3

**Comment by ghiculescu** [comment]
Red test is buildkite failing intermittently.

**Pull request merged**